Mathlib/Algebra/Order/Floor.lean

```diff
@@ -435,6 +435,20 @@ theorem ceil_add_le (a b : α) : ⌈a + b⌉₊ ≤ ⌈a⌉₊ + ⌈b⌉₊ := b
   rw [ceil_le, Nat.cast_add]
   gcongr <;> apply le_ceil
 
+variable [Sub α] [OrderedSub α] [ExistsAddOfLE α]
+
+@[simp] lemma ceil_sub_natCast (a : α) (n : ℕ) : ⌈a - n⌉₊ = ⌈a⌉₊ - n := by
```

Comment:
Is there a version of this with `n - a`?

Reply:
Probably, but I failed to do it in any reasonable time. Here's my best attempt:
```
@[simp] lemma ceil_natCast_sub (ha : 0 ≤ a) (n : ℕ) : ⌈n - a⌉₊ = n - ⌊a⌋₊ := by
  obtain hna | han := le_total ↑n a
  · rwa [ceil_eq_zero.2 (tsub_nonpos_of_le hna), eq_comm, tsub_eq_zero_iff_le, Nat.le_floor_iff]
    exact n.cast_nonneg.trans hna
  obtain rfl | hn := eq_or_ne n 0
  · simp [*]
  refine eq_tsub_of_add_eq <| ?_
  rw [← ceil_add_natCast <| le_tsub_of_add_le_left <| (add_zero a).trans_le han, ceil_eq_iff hn,
    add_comm, ← add_tsub_assoc_of_le han, add_comm]
  sorry
```